scipy/linalg/tests/test_solvers.py

```diff
@@ -513,33 +512,36 @@ def test_solve_discrete_are():
     # relaxed tolerance in gh-18012 after bump to OpenBLAS
     max_tol[11] = 2.5e-13
 
-    def _test_factory(case, atol):
+    @pytest.mark.parametrize("j, case", enumerate(cases))
+    def test_solve_discrete_are(self, j, case):
         """Checks if X = A'XA-(A'XB)(R+B'XB)^-1(B'XA)+Q) is true"""
         a, b, q, r, knownfailure = case
         if knownfailure:
             pytest.xfail(reason=knownfailure)
 
+        atol = self.max_tol[j]
+
         x = solve_discrete_are(a, b, q, r)
-        res = a.conj().T.dot(x.dot(a)) - x + q
-        res -= a.conj().T.dot(x.dot(b)).dot(
-                    solve(r+b.conj().T.dot(x.dot(b)), b.conj().T).dot(x.dot(a))
-                    )
+        bH = b.conj().T
+        xa, xb = x @ a, x @ b
+
+        res = a.conj().T @ xa - x + q
+        res -= a.conj().T @ xb @ (solve(r + bH @ xb, bH) @ xa)
```

Comment:
may as well have `aH` too while we're at it?

Reply:
Good point. How about I pick this up in the next PR? Just to avoid flushing the CI.